charts/release/templates/_deployment.tpl

```diff
@@ -0,0 +1,25 @@
+{{- define "env.deployment" }}
+
+{{- /* Prepare versions for simpler processing */}}
+{{- $versions := include "normalize.versions.deployment" . | mustFromJson }}
+
+{{- range $i, $v := $versions }}
+
+{{- /* Deployment */}}
+{{ include "env.deployment.version.deployment" $v }}
+---
+{{- /* Service */}}
+{{ include "env.deployment.version.service" $v }}
+---
+{{- end }} {{- /* range $i, $v := $versions */}}
+
+{{- /* routemap (and other strategy specific objects) */}}
+{{- if not .Values.application.strategy }}
+{{ include "env.deployment-istio.none" . }}
+{{- else if eq "none" .Values.application.strategy }}
+{{ include "env.deployment-istio.none" . }}
+{{- else }}
+{{- printf "unknown or invalid application strategy (%s) for environment (%s)" .Values.application.strategy .Values.environment | fail }}
```

Comment:
It's not clear to me what's going on here. Why is the only acceptable strategy here `none`?

Reply:
This is the scenario where we have an application made up of a Deployment and a Service but do not have Istio. As a consequence, we cannot implement blue-green or canary. Only none is valid. This is what is used for A/B/n.